### PR TITLE
ROC-1105: Request tracing

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA

https://tools.hmcts.net/jira/browse/ROC-1105

### Change description

Enabled support for request tracing tags and provides utilities for node application to enable it on their side.

### CODE REVIEW

File: log/tracing/middleware.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const uuid = require('uuid')
+
+const { REQUEST_ID_HEADER, ROOT_REQUEST_ID_HEADER } = require('./headers')
+
+function requestTracingMiddleware (req, res, next) {
+  const id = uuid()
+  req.headers[REQUEST_ID_HEADER] = id
+  req.headers[ROOT_REQUEST_ID_HEADER] = id
```
Comment (timja-kainos) [inline log/tracing/middleware.js]: shouldn't these check if they are already present?
just because it is node doesn't mean it is the root request?

Comment (rlewan) [reply log/tracing/middleware.js]: Fair point 👍

File: log/tracing/localStorage.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const { createNamespace } = require('continuation-local-storage')
+
+const INITIAL_REQUEST = 'initialRequest'
+
+const clsNamespace = createNamespace('uk.gov.hmcts.tracing.localStorage')
```
Comment (timja-kainos) [inline log/tracing/localStorage.js]: maybe under the `reform` namespace?

Comment (rlewan) [reply log/tracing/localStorage.js]: Yeah I wasn't sure about that. E.g. in Claim Store we don't reference Reform in out Java package structure.

Comment (rlewan) [reply log/tracing/localStorage.js]: That's why I omitted it here as well.

Comment (timja-kainos) [reply log/tracing/localStorage.js]: we usually put `reform` in for common components and libraries that have no true owner and are shared by everyone.

File: log/express.js [hunk 3]
```diff
@@ -25,33 +27,35 @@ class AccessLogger {
...
+      message: this.formatter(req, res),
+      requestId: req.headers[REQUEST_ID_HEADER],
+      originRequestId: req.headers[ORIGIN_REQUEST_ID_HEADER],
+      rootRequestId: req.headers[ROOT_REQUEST_ID_HEADER]
     });
   }
 
   level(req, res) {
-    return this.userLevel(this.logger, req, res) || this.defaultLevel(req, res);
+    return this.userLevel(this.logger, req, res);
```
Comment (yarpo) [inline log/express.js]: do we use `;`?

File: log/tracing/localStorage.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const { createNamespace } = require('continuation-local-storage')
```
Comment (yarpo) [inline log/tracing/localStorage.js]: and no ';' here? Is there any reason?

Comment (rlewan) [reply log/tracing/localStorage.js]: This module was initially written by a different team and they had a convention for using semicolons. I've added new code using our style, now it's mixed so I'm not entirely happy with it but I think it's possible we'll be able to remove all semicolons later on.

**Pull request merged**